WPFHospitalEditor/HospitalMap.xaml.cs

```diff
@@ -28,47 +29,56 @@ public partial class HospitalMap : Window
         IDoctorServerController doctorServerController = new DoctorServerController();
         ISchedulingServerController schedulingController = new SchedulingServerController();
 
-        public static Canvas canvasHospitalMap;  
+        public static Canvas canvasHospitalMap;
         public static Role role;
         public static List<MapObject> searchResult = new List<MapObject>();
         public static List<EquipmentDto> equipmentSearchResult = new List<EquipmentDto>();
         public static List<MedicationDto> medicationSearchResult = new List<MedicationDto>();
         public static List<RecommendationDto> appointmentSearchResult = new List<RecommendationDto>();
-        private const int regularExaminationDepartment = 1;
 
         public HospitalMap(List<MapObject> allMapObjects, Role role)
-        {                     
+        {
             InitializeComponent();
-            setMapObjectTypeComboBox();
-            setEquipmentTypeComboBox();
-            setMedicationNameComboBox();
-            setDoctorNameComboBox();
-            setNonSelectedComboBoxItem();
-            CanvasService.addObjectToCanvas(mapObjectController.getOutterMapObjects(), canvas);
+            SetMapObjectTypeComboBox();
+            SetEquipmentTypeComboBox();
+            SetMedicationNameComboBox();
+            SetDoctorNameComboBox();
+            SetSpecialistNameComboBox();
+            SetSpecialistEquipmentComboBox();
+            SetNonSelectedComboBoxItem();
```

Comment:
You get all values for different combobox type in beginning, also every time you change search value in specific tab you will call it again. Problem with that is that you will communicate with repositories more often then you should do. That slows application. Also, you will get values for combobox which you probably won't use. Better way to solve this problem is to call specific function for setting combobox values only when you access tab that contains that combobox. Then  when you want to filter you could get all values **from combobox** and filter so they match searching parameters.

Reply:
I think that these methods will be called two times on program start. First time from text changed method in initialization, and second time here. You should check that.

Reply:
Fixed. Now comboboxes are filled when specific tab is selected.